### PR TITLE
Add popover

### PR DESCRIPTION
**Pull request by Pante**
**Describe the changes**
Add `FPopover`.

**Checklist**
- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md).
- [x] I have included the relevant unit/golden tests.
- [x] I have included the relevant samples.
- [x] I have updated the documentation accordingly.
- [x] I have updated the [CHANGELOG.md](../forui/CHANGELOG.md) if necessary.

### CODE REVIEW

File: forui/lib/src/widgets/popover.dart [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                offset: Offset(0, 4),
+                blurRadius: 6,
+                spreadRadius: -1,
+              ),
+              BoxShadow(
+                color: Color(0x1a000000),
+                offset: Offset(0, 2),
+                blurRadius: 4,
+                spreadRadius: -2,
+              ),
```
Comment (kawaijoe) [inline forui/lib/src/widgets/popover.dart:248]: Should we add this into `FPopoverStyle`?

File: docs/pages/docs/popover.mdx [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+      @override
+      void initState() {
+        super.initState();
+        controller = FPopoverController(vsync: this);
+      }
+
+      @override
+      Widget build(BuildContext context) => FPopover(
+            controller: controller,
+            follower: (context, style, _) => Padding(
```
Comment (kawaijoe) [inline docs/pages/docs/popover.mdx:36]: Are we still using `follower`?

File: forui/lib/src/widgets/popover.dart [hunk 3]
```diff
@@ -0,0 +1,283 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+
+/// A controller that controls whether a [FPopover] is shown or hidden.
+final class FPopoverController extends ChangeNotifier {
```
Comment (kawaijoe) [inline forui/lib/src/widgets/popover.dart:10]: Might be missing something here, but do we need `ChangeNotifier`? `notifyListeners()` doesn't seem to be called anywhere.

File: forui/lib/src/widgets/popover.dart [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+  @useResult
+  FPopoverStyle copyWith({
+    BoxDecoration? decoration,
+    EdgeInsets? margin,
+    EdgeInsets? padding,
+  }) =>
+      FPopoverStyle(
+        decoration: decoration ?? this.decoration,
+        padding: padding ?? this.padding,
+      );
```
Comment (kawaijoe) [inline forui/lib/src/widgets/popover.dart:263]: Redundant `margin` property

File: forui/test/golden/popover/zinc-light-shown.png
Comment (kawaijoe) [inline forui/test/golden/popover/zinc-light-shown.png:1]: Does the shadow look a bit strong/non fussy?

File: forui/test/src/widgets/popover/popover_test.dart [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+      expect(find.text('follower'), findsOneWidget);
+
+      await tester.tap(find.text('target'));
+      await tester.pumpAndSettle();
+
+      expect(find.text('follower'), findsNothing);
+    });
+
+    testWidgets('', (tester) async {});
```
Comment (kawaijoe) [inline forui/test/src/widgets/popover/popover_test.dart:102]:
```suggestion
```

**Review by kawaijoe — CHANGES_REQUESTED** [review]
Also, should we add `Overlay` to `FTheme` as a temporary fix.

File: forui/test/golden/popover/zinc-light-shown.png
Comment (Pante) [reply forui/test/golden/popover/zinc-light-shown.png:1]: The shadow looks stronger as the popover is smaller.  Ideally the shadow is calculated depending on the size of the popover but we currently don't have the foundation for that.

File: forui/test/src/widgets/popover/popover_test.dart [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+import 'package:forui/forui.dart';
+import '../../test_scaffold.dart';
+
+void main() {
+  group('FPopover', () {
+    testWidgets('tap outside hides popover', (tester) async {
+      final controller = FPopoverController(vsync: const TestVSync());
+
+      await tester.pumpWidget(
+        MaterialApp(
```
Comment (kawaijoe) [inline forui/test/src/widgets/popover/popover_test.dart:14]: Should we remove `MaterialApp`? To ensure it works stand alone.

Comment (Pante) [reply forui/test/src/widgets/popover/popover_test.dart:14]: It'll be addressed in the tooltip PR

**Pull request merged**